synthesizer/src/vm/helpers/committee.rs

```diff
@@ -31,41 +31,69 @@ use std::str::FromStr;
 use rayon::prelude::*;
 
 /// Returns the committee given the committee map from finalize storage.
-pub fn committee_map_into_committee<N: Network>(
+pub fn committee_and_delegated_maps_into_committee<N: Network>(
     starting_round: u64,
     committee_map: Vec<(Plaintext<N>, Value<N>)>,
+    delegated_map: Vec<(Plaintext<N>, Value<N>)>,
 ) -> Result<Committee<N>> {
     // Prepare the identifiers.
-    let microcredits_identifier = Identifier::from_str("microcredits")?;
-    let is_open_identifier = Identifier::from_str("is_open")?;
+    let is_open_identifier: Identifier<N> = Identifier::from_str("is_open")?;
+    let commission_identifier: Identifier<N> = Identifier::from_str("commission")?;
 
     // Extract the committee members.
-    let committee_members = committee_map
+    let committee_members: IndexMap<Address<N>, (u64, bool, u8)> = committee_map
         .iter()
         .map(|(key, value)| {
             // Extract the address from the key.
             let address = match key {
                 Plaintext::Literal(Literal::Address(address), _) => address,
                 _ => bail!("Invalid committee key (missing address) - {key}"),
             };
+
             // Extract the committee state from the value.
-            match value {
+            let (is_open, commission) = match value {
                 Value::Plaintext(Plaintext::Struct(state, _)) => {
-                    // Extract the microcredits from the value.
-                    let microcredits = match state.get(&microcredits_identifier) {
-                        Some(Plaintext::Literal(Literal::U64(microcredits), _)) => **microcredits,
-                        _ => bail!("Invalid committee state (missing microcredits) - {value}"),
-                    };
                     // Extract the is_open flag from the value.
                     let is_open = match state.get(&is_open_identifier) {
                         Some(Plaintext::Literal(Literal::Boolean(is_open), _)) => **is_open,
                         _ => bail!("Invalid committee state (missing boolean) - {value}"),
                     };
+                    // Extract the commission from the value.
+                    let commission = match state.get(&commission_identifier) {
+                        Some(Plaintext::Literal(Literal::U8(commission), _)) => **commission,
+                        _ => bail!("Invalid committee state (missing commission) - {value}"),
+                    };
                     // Return the committee state.
-                    Ok((*address, (microcredits, is_open)))
+                    (is_open, commission)
                 }
                 _ => bail!("Invalid committee value (missing struct) - {value}"),
-            }
+            };
+
+            // Extract the microcredits for the address from the delegated map.
+            let microcredits = delegated_map.iter().find_map(|(delegated_key, delegated_value)| {
```

Comment:
@ljedrz can you assess this for performance implications? (use `test_committee_and_delegated_maps_into_committee` for profiling)

Reply:
That test takes roughly the same amount of time as before; however, due to the scope of these changes it would be best to run a profile in the IsoNet and compare it against one from a run with the current code if we'd like to make sure that there are no regressions.